src/engines/sonic/placeable.cpp

```diff
@@ -53,11 +56,15 @@ namespace Sonic {
 
 Placeable::Placeable(const Aurora::GFF4Struct &placeable) : Object(kObjectTypePlaceable),
 	_typeID(0xFFFFFFFF), _appearanceID(0xFFFFFFFF), _scale(1.0f) {
+	_id = Common::generateIDNumber();
```

Comment:
Hmm, Placeable::load() reads the _id out of the GFF, though. There's probably no reason it has to be this _id, though, so you can probably add a new variable, say _placeableID or something, and assign the value from the GFF there.